File: playground/steps/simplePython/apiEndpoint.step.py
config = {
    "type": "api",
    "name": "Test API Endpoint",
    "emits": ["test"],
    "flows": ["simple-python"],

    "path": "/test-python",
    "method": "POST",
}

async def handler(req, context):
    context.logger.info("this is a test", { "body": req.get("body"), "req": req })

    await context.emit({ "topic": "test", "data": req.get("body") })
    await context.state.set(context.trace_id, "message", "hello world")

    context.logger.info("State set", { "message": "hello world" })

    return {
        "status": 200,
        "body": { "message": "payload processed" }
    }

File: playground/steps/simpleCSharp/api_endpoint_step.cs
using MotiaCSharp.Types;
using MotiaCSharp;

public class ApiEndpointStep
{
    public static MotiaConfig config = new MotiaConfig
    {
        Type = "api",
        Name = "Test C# API Endpoint",
        Emits = new[] { "test" },
        Flows = new[] { "simple-csharp" },
        Path = "/test-csharp",
        Method = "POST"
    };

    public static async Task<object> handler(object req, MotiaContext context)
    {
        context.Logger.Info("Processing C# API step", new { body = req });

        // Set some state
        await context.State.Set(context.TraceId, "processed", "yes");
        context.Logger.Info("State set", new { message = "hello from C#" });

        // Emit an event
        await context.Emit(new MotiaEvent
        {
            Topic = "test",
            Data = req
        });

        // Return response
        return new
        {
            status = 200,
            body = new { message = "C# step processed successfully" }
        };
    }
}
